### ISSUE TITLE
send notifications with patch build results to gerrit/github

### ISSUE THREAD

**Issue by mwasilew** [Linaro/squad#190]
Imported from https://projects.linaro.org/browse/QA-1627
we should have support for having pluggable notification delivery mechanisms. right now we notify by email, but we also need to e.g. notify on a gerrit review. notifying to github pull requests would be nice to have as well.

milosz.wasilewski:
> Dependency issue QA-1443 resolved with resolution Duplicate.

amro.hassaan:
> Hi [~milosz.wasilewski@linaro.org] any idea when this story will be done? it is blocking CTT-424. Thankx

milosz.wasilewski:
> [~amro.hassaan@linaro.org] not soon unless I'm told otherwise. LKFT has the highest priority at the moment and I don't see the way of fitting this feature before New Year.

amro.hassaan:
> ack ..thanks [~milosz.wasilewski@linaro.org].

amro.hassaan:
> [~mike.holmes@linaro.org], CTT discussed this today in TL meeting. QA is very squeezed up until January pretty much. They estimated January as delivery date for this QA story. We added due date here.

**Comment by fboudra** [comment]
This issue is a blocker for https://projects.linaro.org/browse/CTT-424

**Comment by mwasilew** [comment]
our Jenkins build has the following env variables available for github PRs:
```
ghprbActualCommit
ghprbActualCommitAuthor
ghprbActualCommitAuthorEmail
ghprbAuthorRepoGitUrl
ghprbCommentBody
ghprbCredentialsId
ghprbGhRepository
ghprbPullAuthorEmail
ghprbPullAuthorLogin
ghprbPullAuthorLoginMention
ghprbPullDescription
ghprbPullId
ghprbPullLink
ghprbPullTitle
ghprbSourceBranch
ghprbTargetBranch
```

I guess for simplicity we can use the same names in the metadata provided to SQUAD.

**Comment by terceiro** [comment]
this has been implemented in #282, except for the gerrit part. See #307 for that.
